### PR TITLE
feat: key components health check

### PR DESCRIPTION
**Pull request by zzzming**
Since /health is also supported by uvicorn, so until the langflow's router is initialized. `/health` returns the status of uvicorn. That does not reflect the readiness of the langflow. 

This PR introduces a new endpoint `/health_check` that also test key services like chat, session, and variable services that make sure cache , DB, or kubernetes backend are working. It offers more reliable service type of check rather a simple route based `health` check. Only a few selected services to save the health check time.

Once this PR is merged and released, Kubernetes's probe should use the /health_check endpoint.

### CODE REVIEW

File: src/backend/base/langflow/main.py [hunk 1]
```diff
@@ -137,11 +137,8 @@ async def flatten_query_string_lists(request: Request, call_next):
 
         return await call_next(request)
 
-    @app.get("/health")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/main.py:140]: I think we should keep this for now until we update the frontend.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
This looks very good, Ming. Thank you!

File: src/backend/base/langflow/main.py (hunk 1 above)
Comment (zzzming) [reply src/backend/base/langflow/main.py:140]: @ogabrielluiz `/health` is retained. It's just moved here in another file.
https://github.com/langflow-ai/langflow/pull/2510/files#diff-5f75901a0969078305ed5f9087a232612ba48e19ef19a2af437c93268e8cedacR36

File: src/backend/base/langflow/api/health_check_router.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        logger.exception(e)
+
+    try:
+        create_default_folder_if_it_doesnt_exist(session, uuid.UUID(user_id))
+        response.folder = "ok"
+    except Exception as e:
+        logger.exception(e)
+
+    if response.has_error():
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=response.model_dump())
```
Comment (jordanrfrazier) [inline src/backend/base/langflow/api/health_check_router.py:93]: When we dump this, should the parameter include the error? 

e.g.
```
   try:
        create_default_folder_if_it_doesnt_exist(session, uuid.UUID(user_id))
        response.folder = "ok"
    except Exception as e:
        logger.exception(e)
        response.folder = e.message()
```

**Review by jordanrfrazier — APPROVED** [review]

File: src/backend/base/langflow/main.py (hunk 1 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/main.py:140]: Sounds good. Sorry

**Review by ogabrielluiz — APPROVED** [review]

File: src/backend/base/langflow/api/health_check_router.py (hunk 2 above)
Comment (zzzming) [reply src/backend/base/langflow/api/health_check_router.py:93]: Good suggestion. I will do another PR to improve that.

**Review by nicoloboschi — COMMENTED** [review]
1. This portion of the code needs to be changed. This makes the health check not idempotent and not able to run in parallel, which is problematic in many scenarios. For example, in k8s liveness and readiness probe could run in parallel. 

It's ok to write/read to the cache but there's no need to verify the correctness, we should just check the availability. 

```
        await chat.set_cache("health_check", str(test_id))
        if v := await chat.get_cache("health_check"):
            if v.get("result") == str(test_id):
                response.chat = "ok"
            else:
                logger.error("chat service get incorrect value")
```



2. I don't see the value of testing the variable service. If variables are on the db, you're testing the DB (again!) and if they are on k8s it's very expensive and not worth it (if you're on k8s, k8s is up..)

Also you're leaking the variable in case the deletion fails. 

I would completely remove the variable section.


3. Again for the folder section, there's no need to test the DB connectivity again.

**Pull request merged**